crates/primitives/src/genesis.rs

```diff
@@ -0,0 +1,60 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    utils::serde_helpers::deserialize_stringified_u64, Address, Bytes, Header, H256, U256,
+};
+
+/// The genesis block specification.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Genesis {
+    /// The genesis header nonce.
+    #[serde(deserialize_with = "deserialize_stringified_u64")]
+    pub nonce: u64,
+    /// The genesis header timestamp.
+    #[serde(deserialize_with = "deserialize_stringified_u64")]
+    pub timestamp: u64,
+    /// The genesis header extra data.
+    pub extra_data: Bytes,
+    /// The genesis header gas limit.
+    #[serde(deserialize_with = "deserialize_stringified_u64")]
+    pub gas_limit: u64,
+    /// The genesis header difficulty.
+    pub difficulty: U256,
+    /// The genesis header mix hash.
+    pub mix_hash: H256,
+    /// The genesis header coinbase address.
+    pub coinbase: Address,
+    /// The genesis state root.
+    pub state_root: H256,
```

Comment:
We should remove this and calculate it to be more in line with Geth. You don't need to address it in this PR. If you don't, I would appreciate if you opened up an issue for it afterwards so we can follow up :smile:

Reply:
I created the issue #896